### DISCUSSION TITLE
Getting callback from terminal in extension

### DISCUSSION THREAD

**Discussion by amitjoshi438** [microsoft/vscode-discussions#395] [Extension Development QnA] [answered]
I am working on a VSCode desktop extension for our product and have a use case where I want to get a call back/read data from terminal provided by extension after running a command using sendText(). I went through API docs but was not able to find something related, I came across API window.onDidWriteTerminalData() but now its deprecated. 
Is there any way to achieve this?
Thanks

**Comment by isidorn** [comment]
@tyriar would know best

**Comment by Tyriar** [comment, marked as answer]
You cannot do this currently, the feature is tracked in https://github.com/microsoft/vscode/issues/145234

**Reply by amitjoshi438** [reply]
Okay, it would be great if you can suggest any workaround  to achieve this?

**Reply by Tyriar** [reply]
The only workaround currently is wrapping the program you run in another one you control, such that you can intercept the data written.

**Reply by bwateratmsft** [reply]
In the Docker extension, we've used [`CustomExecution`](https://code.visualstudio.com/api/references/vscode-api#CustomExecution) to run processes with visibility into output. IMO that's probably easier than a wrapper program.

**Reply by Tyriar** [reply]
The `Pseudoterminal` API is similar to that. The problem then is how to launch the pty as depending upon the out of the box node-pty may break when that module is updated and including it as a dependency has challenges wrt platform-specific binaries.

**Reply by amitjoshi438** [reply]
Is using 'child_process' module of Node.js to spawn a new terminal will be a good approach?

**Reply by Tyriar** [reply]
Doing it that way you won't get color, properly sized output or proper interactivity.
